irods/manager/data_object_manager.py

```diff
@@ -17,12 +24,41 @@
 import irods.keywords as kw
 import irods.parallel as parallel
 from irods.parallel import deferred_call
-import six
-import ast
-import json
-import logging
 
+logger = logging.getLogger(__name__)
+
+_update_types = [
+    (typing.Callable, (lambda _:_))  # bare callables in the updatables list should be left as they are
+]
+
+_update_fns = weakref.WeakKeyDictionary()
+
+def register_update_instance(obj_, transform_):
+    _update_fns[obj_] = transform_
+
+def register_update_type(type_ , transform_):
+    global _update_types
+    _update_types.insert(0, (type_,transform_))
+    _update_types = list((k,v) for k,v in collections.OrderedDict(_update_types).items() if v is not None)
 
+def unregister_update_type(type_):
+    register_update_type(type_, None)
+
+def do_progress_updates(updatables, n, logging_function = logger.warning):
+    if not isinstance(updatables, (list,tuple)):
+        updatables = [updatables]
+    for obj in updatables:
+        stored_func = _update_fns.get(obj)
+        if stored_func:
+            stored_func(n)
+        else:
+            # search based on type
+            for cl,func in _update_types:
```

Comment:
Can we add some more comments and/or docstrings for these functions? Consider making the variable names with more letters. It will surely help future readers, and me. :)

Reply:
Improved this. Let me know if more is needed. Or I can resolve tomorrow after short standup revisit & discussion.

Reply:
Yes, seems better. Thanks